LDKNodeMonday/Service/Lightning Service/LightningNodeService.swift

```diff
@@ -11,7 +11,23 @@ import SwiftUI
 import os
 
 class LightningNodeService {
-    static var shared: LightningNodeService = LightningNodeService()
+    private static let lock = NSLock()
+    private static var _shared: LightningNodeService?
+    static var shared: LightningNodeService {
+        get {
+            lock.lock()
+            defer { lock.unlock() }
+            if _shared == nil {
+                _shared = LightningNodeService()
+            }
+            return _shared!
+        }
+        set {
+            lock.lock()
+            _shared = newValue
+            lock.unlock()
+        }
+    }
```

Comment:
What was the reason again that the `static var shared: LightningNodeService = LightningNodeService()` didn't work for this switching network feat?

Reply:
If it never changes the only way to restart the node is to kill the app and restart (manually).
(The extra locking logic was suggested by code rabbit)

Reply:
ah yes that's right!

I've got a suggestion for something that is a bit more modern way of handling things here, I tested and it works, but I'll just save that for a follow up PR since this works just fine at the moment.

Thanks!